src/services/database.js

```diff
@@ -0,0 +1,56 @@
+var sqlite3 = require("sqlite3");
+
+module.exports = function(method,path){
+  if (method === "sqlite") {
+    return SqliteMethod(path);
+  }
+
+  throw "Unrecognized database method: " + method;
+};
+
+function SqliteMethod(path) {
+  var db = new sqlite3.Database(path);
+  db.serialize();
+
+  var refreshTokenColumns = [
+    "client_id TEXT",
+    "refresh_token TEXT"
+  ];
+  var accessTokenColumns = [
+    "access_token TEXT",
+    "iat INTEGER",
+    "exp INTEGER"
+  ];
+
+  db.run("CREATE TABLE IF NOT EXISTS refresh_token (" +
+         refreshTokenColumns.join(", ") + ")");
+  db.run("CREATE TABLE IF NOT EXISTS access_token (" +
+         accessTokenColumns.join(", ") + ")");
+
+  return {
+    run: function () {
+      var args = Array.from(arguments);
+
+      return new Promise(function (resolve, reject) {
+        args.push(function (err) {
+          (err) ? reject(err) : resolve(this);
```

Comment:
From the sqlite3 docs (https://github.com/mapbox/node-sqlite3/wiki/API#databaserunsql-param--callback) it was intended to be the statement object of the query, but I'm pretty sure I haven't tested it, and looking at it now `this` probably got rebound once inside the Promise. I think I'll change that to `resolve()` since nothing is using that statement object.